### PR TITLE
meson: get soversion right

### PR DESCRIPTION
**Pull request by shawnl**
this is necessary to link correctly

### CODE REVIEW

File: contrib/meson/meson.build [hunk 1]
```diff
@@ -40,6 +40,9 @@ libzstd_includes = [include_directories(common_dir, dictbuilder_dir, compress_di
 
 legacy = get_option('legacy_support')
 if legacy != 'false'
+    if legacy == 'true'
+        legacy = '0'
```
Comment (Cyan4973) [inline contrib/meson/meson.build]: Could you describe what this statement is expected to achieve ?

Comment (shawnl) [reply contrib/meson/meson.build]: it allos you to do "meson configure -Dlegacy=true" and have it equivalent to make -DZSTD_LEGACY_SUPPORT=0 (which compiles in all legacy options). That 0 does that is not so clear.

Comment (shawnl) [reply contrib/meson/meson.build]: Not that I see the documentation I see this is totally wrong, I will update the patch

Comment (Cyan4973) [reply contrib/meson/meson.build]: You probably want a positive value >= 1 here

**Comment by shawnl** [comment]
fixed.

**Pull request merged**